module_24_Python/6_abstract_class.py

```diff
@@ -0,0 +1,88 @@
+"""
+Задача 6. Абстрактный класс
+Вы работаете в компании, занимающейся разработкой программного обеспечения для
+архитектурных проектов. Вам необходимо разработать программу для расчёта площади
+различных геометрических фигур, таких как круги, прямоугольники и треугольники.
+
+Задача
+Создайте:
+- класс Shape, который будет базовым классом для всех фигур и будет хранить пустой метод area,
+    который наследники должны переопределить;
+- класс Circle;
+- класс Rectangle;
+- класс Triangle.
+Классы Circle, Rectangle и Triangle наследуют от класса Shape и реализуют метод
+для вычисления площади фигуры.
+Дополнительно: изучите информацию о работе с абстрактными классами.
+
+На основе этой информации сделайте так, чтобы:
+- Нельзя было создавать объекты класса Shape.
+- Наследники класса Shape переопределяли его метод area,
+чтобы объекты этих классов можно было использовать.
+"""
+
+from abc import ABC, abstractmethod
+from math import pi
+
+
+class Shape(ABC):
+
+    def __init__(self, name: str):
+        self.name = name
+
+    @abstractmethod
+    def area(self):
+        pass
+
+
+class Circle(Shape):
+
+    def __init__(self, name: str, r: float):
+        super().__init__(name)
+        self.radius = r
+        self.pi = pi
+
+    def area(self):
+        return pi * self.radius * self.radius
+
+    def __str__(self):
+        return f'{self.name} area is {self.area()}'
+
+
+class Rectangle(Shape):
+
+    def __init__(self, name: str, length: float, width: float):
+        super().__init__(name)
+        self.length = length
+        self.width = width
+
+    def area(self):
+        return self.length * self.length
+
+    def __str__(self):
+        return f'{self.name} area is {self.area()}'
+
+
+class Triangle(Shape):
+
+    def __init__(self, name: str, length: float, base):
```

Comment:
Nobody considers triangle as a combination of base and height
Better create with three sides and calculate area with Heron's formula

Reply:
done